src/ore/src/tracing/mod.rs

```diff
@@ -368,7 +369,35 @@ where
     let stack = stack.with(otel_layer);
     #[cfg(feature = "tokio-console")]
     let stack = stack.with(tokio_console_layer);
-    let stack = stack.with(sentry_tracing::layer().event_filter(sentry_event_filter));
+    let stack = stack.with(
+        sentry_tracing::layer()
+            .event_filter(sentry_event_filter)
+            // WARNING, ENTERING THE SPOOKY ZONE
+            //
+            // While sentry provides an event filter above that maps events to types of sentry events, its `Layer`
+            // implementation does not participate in `tracing`'s level-fast-path implementation, which depends on
+            // a hidden api (<https://docs.rs/tracing-subscriber/latest/src/tracing_subscriber/layer/mod.rs.html#864-867>)
+            // which is primarily manged by filters (like below). The fast path skips verbose log
+            // (and span) levels that no layer is interested by reading a single atomic. Usually, not implementing this
+            // api means "give me everything, including `trace`, unless you attach a filter to me.
+            //
+            // The curious thing here (and a bug in tracing) is that _some configurations of our layer stack above_,
+            // if you don't have this filter can cause the fast-path to trigger, despite the fact
```

Comment:
Ah, so the fact that the fast path was _ever_ enabled was the bug? Wild.

Reply:
YES